frontend/device/android/device.lua

```diff
@@ -387,6 +387,14 @@ end
 
 function Device:showLightDialog()
     local usleep = require("ffi/util").usleep
+    local old_is_fl_sw_on = self.powerd.is_fl_sw_on
+    if not self.powerd.is_fl_sw_on then
+        android:enableFrontlightSwitch()
+        self.powerd.is_fl_sw_on = true
+    end
+
+    self.powerd:setIntensityHW(self.powerd.fl_intensity)
```

Comment:
I don't understand this one. Why we set fl intensity before showing the dialog?

Reply:
Line393: If HardwareSwitch is, the dialog could not change the intensity.

Line 396: If the screen was off, but we want to see the native dialog (in the dark) witch the intensity it had before the toggle.

Reply:
sorry, don't follow. On most android devices this will set intensity to itself, and this is wrong.